source/s3_auto_ranged_put.c

```diff
@@ -36,6 +36,43 @@ static void s_s3_auto_ranged_put_request_finished(
     struct aws_s3_request *request,
     int error_code);
 
+/**
+ * Loads the persistable state used to resume an upload that was previously paused.
+ */
+static int s_load_persistable_state(
+    struct aws_allocator *allocator,
+    struct aws_s3_auto_ranged_put *auto_ranged_put,
+    struct aws_s3_meta_request_persistable_state *persistable_state) {
+
+    auto_ranged_put->synced_data.num_parts_sent = 0;
+    auto_ranged_put->synced_data.num_parts_completed = 0;
+    auto_ranged_put->synced_data.create_multipart_upload_sent = true;
+    auto_ranged_put->synced_data.create_multipart_upload_completed = true;
+    auto_ranged_put->upload_id = aws_string_new_from_string(allocator, persistable_state->multipart_upload_id);
+
+    struct aws_array_list *persisted_etag_list = &persistable_state->etag_list;
+
+    for (size_t etag_index = 0; etag_index < aws_array_list_length(persisted_etag_list); etag_index++) {
+        struct aws_string *etag = NULL;
+        aws_array_list_get_at(&auto_ranged_put->synced_data.etag_list, &etag, etag_index);
+        if (etag != NULL) {
```

Comment:
I don't think there is a case the etag is not NULL. The etag list is just created with empty instead.